File: Models/Tenant/Project/List.cs
using System;
using System.Collections.Generic;
using Warehouse.Models.Joins;

namespace Warehouse.Models
{
    public class List
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Deadline { get; set; }
        public DateTime Created { get; set; }

        // many to one project
        public Project Project { get; set; }
        
        // many to one jobs
        public IList<Job> Jobs { get; set; }
        
        // many to many users
        public IList<ListEmployment> Employments { get; set; }
    }
    
    public class CreateList
    {
        public List List { get; set; }
        public Guid ProjectId { get; set; }
        public IList<User> Employments { get; set; }
    }
    
    public class ListViewModel
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public DateTime Deadline { get; set; }
        public DateTime Created { get; set; }

        public IList<JobListViewModel> Jobs { get; set; }
        public IList<ListEmployment> Employments { get; set; }
    }
}

File: ClientApp/src/app/Models/List.ts
import {Project} from "./Project";
import {Job} from "./Job";
import {User, UserId} from "./User";

export interface List
{
  id?:string;
  name:string;
  deadline?:Date;
  created?: Date;
  description: string;

  project?: Project;
  jobs?: Job[];
  employments?: UserId[];
}

export interface CreateList
{
  list: List;
  projectId: string;
  employments: User[];
}